controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1168,6 +1169,10 @@ func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner,
 		if err != nil {
 			return actionError{errors.Wrap(err, "could not save the host provisioning settings")}
 		}
+		if hfc != nil {
+			info.log.Info("saving hostfirmwarecomponents updates into status")
+			saveHostFirmwareComponents(hfc, info)
```

Comment:
This updates the status when manual cleaning starts, but what if it fails?

I suspect we only want to update this after cleaning has succeeded, but we also have to think about what happens if updating the status fails.

Reply:
right!, yeah I agree we want to only update if it succeeded.
You mean updating the status in case an error happened during cleaning or what?

Reply:
This is how I handled status updates - https://github.com/openshift/baremetal-operator/blob/master/controllers/metal3.io/baremetalhost_controller.go#L1571

Reply:
Thanks @hroyrh

Reply:
I mean if the call to Update() fails, do we get the chance to try again? Or will we restart the whole manual cleaning process or something weird like that?

Reply:
But we don't have a  way to remember that the Update failed in the last Reconcile, right ? So, do we have to check if the config changes were already applied before, by fetching the current Ironic node and if yes - then, simply run Update again, rather than the whole manual cleaning process as you mentioned ?

Reply:
One way, could be  like @hroyrh mentioned I think...
We can check if the information ironic provides about Components would be different (if there was a Firmware Update something would be different in the ironic DB)
Does it makes sense?